File: app/src/main/java/com/bignerdranch/android/mediafiles/ui/recycleView/MediaFileAdapter.java
package com.bignerdranch.android.mediafiles.ui.recycleView;

import android.util.Log;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.paging.PagedListAdapter;
import androidx.recyclerview.widget.DiffUtil;

import com.bignerdranch.android.mediafiles.discovery.model.MediaFile;

public class MediaFileAdapter extends PagedListAdapter<MediaFile, MediaFileViewHolder> {

    private static final String TAG = "LocalItemAdapter";

    public MediaFileAdapter() {
        super(new DiffUtil.ItemCallback<MediaFile>() {
            @Override
            public boolean areItemsTheSame(@NonNull final MediaFile oldItem, @NonNull final MediaFile newItem) {
                return oldItem.getId() == newItem.getId();
            }

            @Override
            public boolean areContentsTheSame(@NonNull final MediaFile oldItem, @NonNull final MediaFile newItem) {
                return oldItem.equals(newItem);

            }
        });
    }

    @NonNull
    @Override
    public MediaFileViewHolder onCreateViewHolder(@NonNull final ViewGroup parent, final int viewType) {
        return MediaFileViewHolder.createFromParent(parent);
    }

    @Override
    public void onBindViewHolder(@NonNull final MediaFileViewHolder holder, final int position) {
        final @Nullable MediaFile item = getItem(position);

        if (item == null) {
            Log.w(TAG, "Null LocalItem at position: " + position);
            return;
        }

        holder.idView.setText(String.valueOf(item.getId()));
        holder.pathView.setText(item.getPath() != null ? item.getPath() : "null");
        holder.unifiedIdView.setText(String.valueOf(item.getMediaStoreId()));
        holder.addedView.setText(String.valueOf(item.getDateAdded()));
        holder.takenView.setText(String.valueOf(item.getDateAdded()));
        holder.modifiedView.setText(String.valueOf(item.getSize()));
    }
}

File: app/src/main/java/com/bignerdranch/android/mediafiles/ui/recycleView/MediaFileAdapter.kt
package com.bignerdranch.android.mediafiles.ui.recycleView

import android.view.ViewGroup
import androidx.paging.PagedListAdapter
import androidx.paging.PagingDataAdapter
import androidx.recyclerview.widget.DiffUtil
import com.bignerdranch.android.mediafiles.discovery.model.MediaFile
import com.bignerdranch.android.mediafiles.util.log.Logger
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class MediaFileAdapter(val logger: Logger) : PagingDataAdapter<MediaFile, MediaFileViewHolder>(object : DiffUtil.ItemCallback<MediaFile>() {
    override fun areItemsTheSame(oldItem: MediaFile, newItem: MediaFile): Boolean {
        return oldItem.id == newItem.id
    }

    override fun areContentsTheSame(oldItem: MediaFile, newItem: MediaFile): Boolean {
        return oldItem.id == newItem.id
    }
}) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MediaFileViewHolder {
        return MediaFileViewHolder.createFromParent(parent)
    }

    override fun onBindViewHolder(holder: MediaFileViewHolder, position: Int) {
        val item = getItem(position)
        if (item == null) {
            logger.w(TAG, "Null LocalItem at position: $position")
            return
        }
        holder.idView.text = item.id.toString()
        holder.pathView.text = item.path?.substringAfterLast("/") ?: "null"
        holder.unifiedIdView.text = item.mediaStoreId.toString()
        holder.addedView.text = timeToDate(1000 * item.dateAdded)
        holder.takenView.text = timeToDate(item.dateTaken)
        holder.modifiedView.text = timeToDate(1000 * item.dateModify)
    }

    private fun timeToDate(time: Long): String {
        val sdf = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US)
        return sdf.format(Date(time))
    }

    companion object {
        private const val TAG = "LocalItemAdapter"
    }
}
